File: spring-boot-java/src/main/java/com/example/hib/Product.java
package com.example.hib;

import jakarta.persistence.*;

/**
 * JPA entity with a private lazy field.
 * Bytecode enhancement (hibernate-maven-plugin) injects $$_hibernate_* machinery.
 */
@Entity
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Basic(fetch = FetchType.LAZY)
    private String description;

    protected Product() {
    }

    public Product(String name, String description) {
        this.name = name;
        this.description = description;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Product{id=" + id + ", name='" + name + "', description='" + description + "'}";
    }
}

File: spring-boot-kotlin/src/main/kotlin/com/example/hib/Product.kt
package com.example.hib

import jakarta.persistence.*

/**
 * Kotlin entity for Hibernate 7 bytecode enhancement.
 * IMPORTANT: the class must be `open` and fields `var` (Kotlin classes are `final`
 * by default, and the enhancer cannot weave into a final class).
 */
@Entity
@Table(name = "product_k")
open class Product(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    var name: String,

    @Basic(fetch = FetchType.LAZY)
    var description: String?,
)